app/src/services/storage.js

```diff
@@ -193,10 +193,14 @@ const objectStorageService = {
    * @function headBucket
    * Checks if a bucket exists and if the S3Client has correct access permissions
    * @param {string} [bucketId] Optional bucketId
+   * @param {Object} [bucketSettings] Optional bucketSettings for checking new additions/updates
    * @returns {Promise<object>} The response of the head bucket operation
    */
-  async headBucket(bucketId = undefined) {
-    const data = await utils.getBucket(bucketId, true);
+  async headBucket(bucketId = undefined, bucketSettings = null) {
+    // If already providing bucket settings don't go and get them from the DB or configuration
+    const data = bucketSettings
+      ? bucketSettings
+      : await utils.getBucket(bucketId, true);
```

Comment:
I think we should change this function signature to accept a generic options object instead of having two argument types, and then discern/prioritize logic based on the existence/lack of bucketId attribute:

```js
/**
 * @function headBucket
 * Checks if a bucket exists and if the S3Client has correct access permissions
 * You must supply either a `bucketId`, or the full S3 client credentials to test against
 * @param {string} [options.bucketId] Optional bucketId
 * @param {string} [options.accessKeyId] Optional S3 accessKeyId
 * @param {string} [options.bucket] Optional S3 bucket
 * @param {string} [options.endpoint] Optional S3 endpoint
 * @param {string} [options.key] Optional S3 key/prefix
 * @param {string} [options.region] Optional S3 region
 * @param {string} [options.secretAccessKey] Optional S3 secretAccessKey
 * @returns {Promise<HeadBucketCommandOutput>} The response of the head bucket operation
 */
async headBucket(options = {}) {
  const data = options.bucketId
    ? await utils.getBucket(options.bucketId, true)
    : options;
  const params = {
    Bucket: data.bucket,
  };

  return this._getS3Client(data).send(new HeadBucketCommand(params));
}
```